File: Aufgabe1/solutions/1.4/sqrt_lib.py
def sqrt_newton(x):
  EPS = 0.000001
  MI  = 10000

  iter     = 0
  estimate = 1.0

  while (abs(estimate*estimate - x) > EPS and iter < MI):
    estimate = 0.5*(estimate + x/estimate)
    iter = iter + 1

  return estimate
  

File: Aufgabe1/solutions/1.4/sqrt_lib.cpp
#include <cmath>

double sqrt_newton(double x) {
  const double       EPSILON  = 0.000001;
  const unsigned int MAX_ITER = 10000;

  unsigned int iter     = 0;
  double       estimate = 1.0;

  while (::fabs(estimate*estimate - x) > EPSILON && iter < MAX_ITER) {
    estimate = 0.5*(estimate + x/estimate);
    ++iter;
  }

  return estimate;
}
